Makefile.include

```diff
@@ -386,14 +386,20 @@ distclean:
 	-@for i in $(USEPKG) ; do "$(MAKE)" -C $(RIOTPKG)/$$i distclean ; done
 	-@rm -rf $(BINDIRBASE)
 
-flash: all $(FLASHDEPS)
+ifneq (,$(filter flash, $(MAKECMDGOALS)))
+  PREFLASHDEPS += all
```

Comment:
Its not really `PREFLASHDEPS` as its also a deps for `flash`.
I do not think it needs to be made generic with the += also.

What we actually mean here is "flash when the firmware is built", by saying `firmware.elf/.bin` as a dependency but we can't right now. Maybe more a `FIRMWARE_IS_BUILT` or something.
Could get some other inputs, I am bad at names.